cartoframes/viz/source.py

```diff
@@ -199,9 +199,8 @@ def _set_source_query(self, dataset, bounds):
 
     def _set_source_geojson(self, dataset, bounds):
         self.type = SourceType.GEOJSON
-        gdf = get_geodataframe(dataset)
-        self.query = get_encoded_data(gdf)
-        self.bounds = bounds or get_bounds(gdf)
+        self.query = get_encoded_data(dataset.geodataframe())
```

Comment:
we can cache `dataset.geodataframe()` to a variable to avoid future duplication of logic.

Reply:
:+1:

Reply:
+1